### ISSUE TITLE
VST2 vs OBS implementation: What are the differences?

### ISSUE THREAD

**Issue by Mxt-dot** [werman/noise-suppression-for-voice#59]
I compared the quality of the noise suppression on Streamlabs/OBS (with RNNoise) to the VST2 plugin in equalizerAPO.

Somehow the Streamlabs OBS one seems to work tremendously better, meaning it does not cut off in the middle of the sentence and in general seems to be very insensitive and accurate.

Do you think this has anything to do with Streamlabs/OBS doing some tinkering on their own - which I currently can't spot from the source code for OBS - or is this some consequence from this being a VST2 plugin? 
Perhaps a different model?

I know you weren't involved in the actual product, but perhaps you can pinpoint me as to how this difference exists.

**Comment by werman** [comment]
I tested OBS and I don't think I could hear any considerable noise suppression in it. It doesn't filter my keyboard at all, and any other loud sound. I don't know about Streamlabs. 

> meaning it does not cut off in the middle of the sentence

This should depend on VAD setting, if your VAD threshold is above zero - the plugin need to detect voice to output anything but silence, so it may cut a start of the words (however it should not cut you in the middle of the sentence, try to lower the threshold). 

The above could be solved with a small additional delay. Probably I need to add such setting.

> Somehow the Streamlabs OBS one seems to work tremendously better

Beside the issue described above, I didn't hear anything sounding much better, if there is such a difference, I'd like to hear it.

**Comment by Mxt-dot** [comment]
Alright so I tried to record a little sample. Keep in mind that the reference audio had some gain so it's a bit more clear to hear. The VST2 one and the Streamlabs OBS one didn't have the gain.
All of those audio files got converted from flv to flac, so hopefully the audio isn't too bad for you to listen to :) 
Personally, I can clearly hear in the vst sample where it cuts off my voice.

I don't know the recommended way of providing some samples, so I uploaded them to a file host.
password: rnnoise

https://sendeyo.com/en/d0c6f4e9f5 this is the reference one without any noise filtering

https://sendeyo.com/en/289aa8c4ec with rnnoise from the VST

https://sendeyo.com/en/0970b968e7 with rnnoise from the Streamlabs OBS option.

I recorded the VST one with VAD = 0 and checked it via the embedded text in equalizerAPO. Or maybe I'm just going crazy, I don't know :D 

Thanks for the quick answer and troubleshooting help so far.
